File: 3712-minimum-cost-to-make-arrays-identical/minimum-cost-to-make-arrays-identical.java
class Solution {
    public long minCost(int[] arr, int[] brr, long k) {
        long ans1 = 0;
        int n=arr.length;

        for(int i=0;i<n;i++)
        {
            ans1 += Math.abs(arr[i]-brr[i]);
        }

        long ans2 = k;
        Arrays.sort(arr);
        Arrays.sort(brr);

        for(int i=0;i<n;i++)
        {
            ans2 += Math.abs(arr[i]-brr[i]);
        }

        return Math.min(ans1,ans2);
        
    }
}

File: 3712-minimum-cost-to-make-arrays-identical/minimum-cost-to-make-arrays-identical.cpp
class Solution {
public:
    long long minCost(vector<int>& arr, vector<int>& brr, long long k) {
        //Don't use k...
        long long ans=0;
        for(int i=0; i<arr.size(); i++) ans+=abs(arr[i]-brr[i]);

        //Use K once...
        sort(arr.begin(), arr.end());
        sort(brr.begin(), brr.end());
        long long ans2=k;
        for(int i=0; i<arr.size(); i++) ans2+=abs(arr[i]-brr[i]);
        return min(ans, ans2);
        
    }
};